### PR TITLE
Temp fix wasFaultInjectedOnService for non-grpc and non-services

### LINKED ISSUE #162: Fix wasFaultInjectedOnService for non-GRPC and non-services.
I would say a temporary fix here is the following:

1. Copy this wasFaultInjectedOnService to Assertions/Grpc.java, so that assumption is fine there.  This method should throw if the format doesn't match a Grpc service method format -- therefore, in this case throw with some sort of exception to indicate not-GRPC.
2. In this method -- and if getServerBackendCanInvokeDirectlyProperty() is true -- throw exception saying that this API is currently not supported generally and recommend the developer use the Grpc variant instead.  This will ensure that the backend Python API keeps working.
3. 
Then, you can test both of these in this test and they should both throw and we keep everything that is working, continues to work.

### PR DESCRIPTION
**Pull request by MichaelA-ops**
Closes #162

### CODE REVIEW

File: src/main/java/cloud/filibuster/junit/assertions/Grpc.java [hunk 1]
```diff
@@ -107,4 +110,28 @@ private static void enableFaultInjection() {
...
+    /**
+     * Determine if a fault was injected during the current test execution.
+     *
+     * Only works if the target service is instrumented using a server instrumentor and reports its name.
+     *
+     * @param serviceName service name, as reported by a server instrumentor.
+     * @return was fault injected
+     */
+    public static boolean wasFaultInjectedOnService(String serviceName) {
+        if (serviceName != null && serviceName.toLowerCase(Locale.ROOT).contains("service")) {
```
Comment (MichaelA-ops) [inline src/main/java/cloud/filibuster/junit/assertions/Grpc.java:123]: Hi @cmeiklejohn , point 1 in #162 says I should throw if the format doesn't match a GRPC service. I am not sure how I could tell from the service name whether it's a GRPC service.

Comment (cmeiklejohn) [reply src/main/java/cloud/filibuster/junit/assertions/Grpc.java:123]: For now, just check */*

File: src/main/java/cloud/filibuster/junit/Assertions.java [hunk 2]
```diff
@@ -172,7 +173,7 @@ public static boolean wasFaultInjected() {
         return wasFaultInjected("/filibuster/fault-injected");
     }
 
-    private static boolean wasFaultInjected(String uri) {
+    public static boolean wasFaultInjected(String uri) {
```
Comment (cmeiklejohn) [inline src/main/java/cloud/filibuster/junit/Assertions.java:176]: Was this a necessary change?

File: src/main/java/cloud/filibuster/junit/assertions/Grpc.java [hunk 3]
```diff
@@ -107,4 +110,28 @@ private static void enableFaultInjection() {
...
+                if (FilibusterCore.hasCurrentInstance()) {
+                    return FilibusterCore.getCurrentInstance().wasFaultInjectedOnService(serviceName);
+                } else {
+                    return false;
+                }
+            } else {
+                return wasFaultInjected("/filibuster/fault-injected/service/" + serviceName);
+            }
+        } else {
+            throw new FilibusterRuntimeException("Service name does not indicate a GRPC service.");
```
Comment (cmeiklejohn) [inline src/main/java/cloud/filibuster/junit/assertions/Grpc.java:134]: Use that unsupported API exception.

File: src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterSyncGetTest.java [hunk 4]
```diff
@@ -67,8 +66,7 @@ public void testRedisSyncGet() {
             testExceptionsThrown.add(t.getMessage());
 
             assertTrue(wasFaultInjected(), "An exception was thrown although no fault was injected: " + t);
-            assertTrue(wasFaultInjectedOnService("io.lettuce.core.api.sync.RedisStringCommands"), "Fault was not injected on the expected Redis service: " + t);
```
Comment (cmeiklejohn) [inline src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterSyncGetTest.java:70]: Can you call this and make sure it throws?

File: src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterSyncGetNonExistingKeyTest.java [hunk 5]
```diff
@@ -68,8 +67,7 @@ public void testRedisSyncGetNonExistingKey() {
             testExceptionsThrown.add(t.getMessage());
 
             assertTrue(wasFaultInjected(), "An exception was thrown although no fault was injected: " + t);
-            assertTrue(wasFaultInjectedOnService("io.lettuce.core.api.sync.RedisStringCommands"), "Fault was not injected on the Redis module: " + t);
```
Comment (cmeiklejohn) [inline src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterSyncGetNonExistingKeyTest.java:71]: Can you call this and make sure it throws?

File: src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterSyncGetDefaultTest.java [hunk 6]
```diff
@@ -67,8 +66,7 @@ public void testRedisSyncGetDefaultTests() {
             testExceptionsThrown.add(t.getMessage());
 
             assertTrue(wasFaultInjected(), "An exception was thrown although no fault was injected." + t);
-            assertTrue(wasFaultInjectedOnService("io.lettuce.core.api.sync.RedisStringCommands"), "Fault was not injected on the Redis module: " + t);
```
Comment (cmeiklejohn) [inline src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterSyncGetDefaultTest.java:70]: Can you call this and make sure it throws?

File: src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterRetryTest.java [hunk 7]
```diff
@@ -50,8 +49,7 @@ public void testRedisRetry() {
             assertEquals("Hello, " + value + "!!", reply.getValue());
         } catch (Throwable t) {
             if (wasFaultInjected()) {
-                assertTrue(wasFaultInjectedOnService("io.lettuce.core.RedisFuture"), "Fault was not injected on the Redis module");
```
Comment (cmeiklejohn) [inline src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterRetryTest.java:53]: Can you call this and make sure it throws?

File: src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterByzantineStringTest.java [hunk 8]
```diff
@@ -58,8 +57,7 @@ public void testRedisByzantineGet() {
         } else {
             actualValues.add(returnVal);
             assertTrue(expectedValues.contains(returnVal), "An unexpected value was returned: " + returnVal);
-            assertTrue(wasFaultInjectedOnService("io.lettuce.core.api.sync.RedisStringCommands"), "Fault was not injected on the Redis module");
```
Comment (cmeiklejohn) [inline src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterByzantineStringTest.java:61]: Can you call this and make sure it throws?

File: src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterByzantineByteArrTest.java [hunk 9]
```diff
@@ -64,8 +63,7 @@ public void testRedisByzantineGet() {
             actualValues.add(returnVal);
             String returnValStr = returnVal == null ? null : new String(returnVal, Charset.defaultCharset());
             assertTrue(expectedValues.contains(returnValStr), "An unexpected value was returned: " + returnValStr);
-            assertTrue(wasFaultInjectedOnService("io.lettuce.core.api.sync.RedisStringCommands"), "Fault was not injected on the Redis module");
```
Comment (cmeiklejohn) [inline src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterByzantineByteArrTest.java:67]: Can you call this and make sure it throws?

File: src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterBuggyFallbackByzantineTest.java [hunk 10]
```diff
@@ -45,8 +44,7 @@ public void testRedisByzantineBuggyFallback() {
             assertEquals(value, reply.getValue());
         } catch (Throwable t) {
             if (wasFaultInjected()) {
-                assertTrue(wasFaultInjectedOnService("io.lettuce.core.api.sync.RedisStringCommands"), "Fault was not injected on the Redis module");
```
Comment (cmeiklejohn) [inline src/test/java/cloud/filibuster/functional/java/redis/JUnitRedisFilibusterBuggyFallbackByzantineTest.java:48]: Can you call this and make sure it throws?

**Review by cmeiklejohn — CHANGES_REQUESTED** [review]

File: src/main/java/cloud/filibuster/junit/assertions/Grpc.java (hunk 1 above)
Comment (MichaelA-ops) [reply src/main/java/cloud/filibuster/junit/assertions/Grpc.java:123]: `wasFaultInjectedOnService` is mostly called with strings in other formats though. For example:

https://github.com/filibuster-testing/filibuster-java-instrumentation/blob/f5437c69529602c9d92e7e9e897e845a91a18784/src/test/java/cloud/filibuster/functional/java/basic/BasicTest.java#L81

**Review by cmeiklejohn — CHANGES_REQUESTED** [review]
Looks good, but I need you to bump the version and the CHANGELOG since you changed a core API.

**Review by cmeiklejohn — APPROVED** [review]

**Pull request merged**